arrows/serialize/protobuf/CMakeLists.txt

```diff
@@ -0,0 +1,56 @@
+# Build / Install plugin for serialization
+
+find_package(Protobuf MODULE REQUIRED)
+
+if(NOT PROTOBUF_FOUND)
+  message(FATAL_ERROR "-- Protobuf is required!")
+endif()
+
+set( headers_public
+  bounding_box.h
+  detected_object_type.h
+  detected_object.h
+  detected_object_set.h
+  )
+
+set( sources
+  bounding_box.cxx
+  detected_object_type.cxx
+  detected_object.cxx
+  detected_object_set.cxx
+  )
+
+link_directories(${PROTOBUF_LIBRARY})
+include_directories(${PROTOBUF_INCLUDE_DIR})
+add_compile_options( -include "${CMAKE_BINARY_DIR}/vital/types/protobuf/vital_types_protobuf_export.h" )
+
+kwiver_add_library( kwiver_serialize_protobuf
+  ${headers_public}
+  ${sources}
+  )
+
+target_link_libraries( kwiver_serialize_protobuf
+  PUBLIC               vital
+                       vital_vpm
+  PRIVATE              vital_algo
+                       vital_exceptions
```

Comment:
I think vital_algo is in an included header, which might require it to be PUBLIC. Hard to follow the paper trail though, could be wrong.

Reply:
Similarly, it looks like vital_vpm's headers aren't included in this libraries heads, which means it probably doesn't need to be a PUBLIC link. The same might apply for the json serialization as well.

Reply:
Changed visibility to be more like other plugin libraries